platform/security/idp/security-idp-plugin/security-idp-plugin-nameidpolicy/src/test/groovy/org/codice/ddf/security/idp/plugin/nameidpolicy/NameIdPolicyPresignPluginSpec.groovy

```diff
@@ -0,0 +1,417 @@
+
+class NameIdPolicyPresignPluginSpec extends Specification {
+    def static INCORRECT_VALUE = "incorrect"
+    def static EXAMPLE_SP_NAME_QUALIFIER = "https://example-ddf-sp.com"
+    def static EXAMPLE_NAME_ID_VALUE = "admin"
+    def static EXAMPLE_EMAIL = "admin@localhost"
+
+    def static NAME_IDENTIFIER_URI =
+            "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"
+    def static EMAIL_URI = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
+
+    def plugin = new NameIdPolicyPresignPlugin()
+
+    Response responseMock
+    NameID nameIdMock
+    Attribute emailAttribute
+
+    def authnRequestMock = Mock(AuthnRequest)
+    def nameIdPolicyMock = Mock(NameIDPolicy)
+    def spMetadataMock = new ArrayList<>()
+    def supportedBindingsMock = new HashSet<>()
+
+    def setupSpec() {
+        OpenSAMLUtil.initSamlEngine()
+    }
+
+    def setup() {
+        def cannedResponse = this.getClass().getResource('/sampleResponse.xml').text
+        Document responseDoc = StaxUtils.read(new ByteArrayInputStream(cannedResponse.getBytes()))
+        XMLObject responseXmlObject = OpenSAMLUtil.fromDom(responseDoc.getDocumentElement())
+        responseMock = (Response) responseXmlObject
+        nameIdMock = responseMock.assertions.first().subject.nameID
+        nameIdMock.setValue(EXAMPLE_NAME_ID_VALUE)
+        nameIdMock.setFormat(NameID.UNSPECIFIED)
+
+        emailAttribute =
+                responseMock.assertions.first().attributeStatements.first().attributes.first()
+
+        authnRequestMock.getNameIDPolicy() >> nameIdPolicyMock
+    }
+
+    @Unroll("null NameIDPolicy Format tests for Format #format")
+    def 'null NameIDPolicy Format tests'() {
+        setup:
+        nameIdMock.setFormat(format)
+        nameIdPolicyMock.getFormat() >> policyFormat
+
+        when:
+        plugin.processPresign(responseMock, authnRequestMock, spMetadataMock, supportedBindingsMock)
+
+        then:
+        if (!isError) {
+            assert (nameIdMock.value == EXAMPLE_NAME_ID_VALUE)
+            return
+        }
+        thrown(UnsupportedOperationException)
+
+        where:
+        policyFormat | format              | isError
+        null         | null                | false
+        null         | NameID.UNSPECIFIED  | false
+        null         | NameID.PERSISTENT   | false
+        null         | NameID.TRANSIENT    | false
+        null         | NameID.X509_SUBJECT | false
+        null         | NameID.EMAIL        | false
+    }
+
+    @Unroll("unspecified NameIDPolicy Format tests for Format #format")
+    def 'unspecified NameIDPolicy Format tests'() {
+        setup:
+        nameIdMock.setFormat(format)
+        nameIdPolicyMock.getFormat() >> policyFormat
+
+        when:
+        plugin.processPresign(responseMock, authnRequestMock, spMetadataMock, supportedBindingsMock)
+
+        then:
+        if (!isError) {
+            assert (nameIdMock.value == EXAMPLE_NAME_ID_VALUE)
```

Comment:
❗️ I would think we'd want to also verify that the nameId's format value is equal to `format`